moto/kinesis/models.py

```diff
@@ -574,6 +576,11 @@ def merge_shards(self, stream_name, shard_to_merge, adjacent_shard_to_merge):
                 record.partition_key, record.data, record.explicit_hash_key
             )
 
+    @paginate(pagination_model=PAGINATION_MODEL)
+    def list_shards(self, stream_name, limit=None, next_token=None):
+        stream = self.describe_stream(stream_name)
+        return [shard.to_json() for shard in stream.shards.values()]
```

Comment:
The model can simply return the entire list of (applicable) items.
The decorator will convert this into a `(curated_item_list, token)`-tuple